### PR TITLE
Remove 'Verify contract details' link on Sig Req screen when 'verifyingContract' is absent

### LINKED ISSUE #17116: [Bug]: Sign Type Data without `verifyingContract` lands to MetaMask Error page
### Describe the bug

**Problem**: when a dapp is using the Sign Type Data functionality and does not set any `verifyingContract` , if I click to Contract Details on MM prompt, I land into the MM encountered an error page.

error-page

https://user-images.githubusercontent.com/54408225/211546491-9f18a147-2719-4c48-a7cc-680543ff0511.mp4

### Steps to reproduce

1. Clone test dapp https://github.com/MetaMask/test-dapp
2. Remove `verifyingContract` fields from the Sign Typed Data (it does not matter which version)
3. Run the test dapp project `npm start`
4. Open localhost:9011
5. Connect MM to the test dapp
6. Click on Sign for the Typed Data version that you modified
7. On MM prompt click on Contract Details -- you land on the Error page

### Error messages or log output

_No response_

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/MetaMask/metamask-extension/issues/17116

## Explanation
When `verifyingContract` is not present in `txData.msgParams.data.domain`, we should not show the navigation link to the verifying contract view. 

Though this piece does not address the issue (With the screen no longer reachable the error mentioned in the issue isn't thrown), `ellipsify()` has been hardened against `undefined` input, with tests added for the function.

### Before
error-page

https://user-images.githubusercontent.com/54408225/211546491-9f18a147-2719-4c48-a7cc-680543ff0511.mp

### After
Screen Shot 2023-01-10 at 4 55 07 PM

## Manual Testing Steps
1. Clone test dapp https://github.com/MetaMask/test-dapp
2. Remove `verifyingContract` fields from the Sign Typed Data (it does not matter which version)
3. Run the test dapp project `npm start`
4. Open localhost:9011
5. Connect MM to the test dapp
6. Click on Sign for the Typed Data version that you modified
7. Confirm that the `Verify contract details` link is not present
8. Repeat steps with `verifyingContract` details present
9. Ensure that the `Verify contract details` link is present on the sig req confirmation screen, and can be clicked through as expected.

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by Charly-1944** [comment]
Ok

**Review by chloeYue — DISMISSED** [review]
QA test OK, LGTM, thanks!

**Review by jpuri — DISMISSED** [review]

File: ui/components/app/signature-request/signature-request.component.js [hunk 1]
```diff
@@ -222,20 +224,22 @@ export default class SignatureRequest extends PureComponent {
...
-          </div>
+                <Typography
+                  variant={TYPOGRAPHY.H7}
+                  color={COLORS.PRIMARY_DEFAULT}
+                >
+                  {this.context.t('verifyContractDetails')}
+                </Typography>
+              </Button>
+            </div>
+          ) : null}
```
Comment (jpuri) [inline ui/components/app/signature-request/signature-request.component.js:242]: Nit: will be nice to have this condition also tested.

Comment (ryanml) [reply ui/components/app/signature-request/signature-request.component.js:244]: @jpuri I have added a test for this. cc: @chloeYue

**Review by jpuri — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
QA test OK, LGTM !

**Pull request merged**